src/engine/controls/bpmcontrol.cpp

```diff
@@ -174,14 +188,104 @@ void BpmControl::slotAdjustBeatsFaster(double v) {
     if (v <= 0) {
         return;
     }
-    adjustBeatsBpm(kBpmAdjustStep);
+
+    const TrackPointer pTrack = getEngineBuffer()->getLoadedTrack();
+    if (!pTrack) {
+        return;
+    }
+    const mixxx::BeatsPointer pBeats = pTrack->getBeats();
+    if (!pBeats) {
+        return;
+    }
+
+    const auto currentPosition = frameInfo().currentPosition.toLowerFrameBoundary();
+
+    auto markers = pBeats->getMarkers();
+    auto markerIt = std::upper_bound(markers.begin(),
+            markers.end(),
+            currentPosition,
+            [](const mixxx::audio::FramePos& position,
+                    const mixxx::BeatMarker& marker) {
+                return position < marker.position();
+            });
+    if (markerIt != markers.begin()) {
+        markerIt--;
+    }
+
+    auto endMarkerBpm = pBeats->getEndMarkerBpm();
+    if (markerIt == markers.end()) {
+        endMarkerBpm += kBpmAdjustStep;
+    } else {
```

Comment:
This is intentional. The last marker's BPM is only changed when the play position is behind the last marker (or if there are no other markers). Otherwise, it increases the beast count (and hence the tempo) of the section delimited by the markers adjacent to the play position.